cmd/evm/internal/t8ntool/execution.go

```diff
@@ -196,6 +196,16 @@ func (pre *Prestate) Apply(vmConfig vm.Config, chainConfig *params.ChainConfig,
 		evm := vm.NewEVM(vmContext, vm.TxContext{}, statedb, chainConfig, vmConfig)
 		core.ProcessBeaconBlockRoot(*beaconRoot, evm, statedb)
 	}
+	if pre.Env.BlockHashes != nil && chainConfig.IsPrague(new(big.Int).SetUint64(pre.Env.Number), pre.Env.Timestamp) {
+		// insert all parent hashes in the contract
+		for i, h := range pre.Env.BlockHashes {
+			n := uint64(i)
+			if n >= pre.Env.Number || n < pre.Env.Number-257 {
```

Comment:
```suggestion
			if n >= pre.Env.Number || pre.Env.Number > (n+params.HistoryServeWindow) {
```
Values were not appearing in the storage after Prague during testing, I think it's because pre.Env.Number underflows on a low block count.
With this change the storage is correctly updated.